chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +934,323 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
             return spend_bundle
         else:
             raise ValueError("Couldn't set DID on given NFT")
+
+    async def mint_from_did(
+        self,
+        metadata_list: List[Dict],
+        target_list: Optional[List[bytes32]] = [],
+        mint_number_start: Optional[int] = 1,
+        mint_total: Optional[int] = None,
+        xch_coins: Optional[Set[Coin]] = None,
+        xch_change_ph: Optional[bytes32] = None,
+        new_innerpuzhash: Optional[bytes32] = None,
+        did_coin: Optional[Coin] = None,
+        did_lineage_parent: Optional[bytes32] = None,
+        fee: Optional[uint64] = uint64(0),
+    ):
+        """
+        Minting NFTs from the DID linked wallet, also used for bulk minting NFTs.
+        - The DID is spent along with an intermediate launcher puzzle which
+          generates a set of ephemeral coins with unique IDs by currying in the
+          mint_number and mint_total for each NFT being minted. These
+          intermediate coins then create the launcher coins for the list of NFTs
+        - The launcher coins are then spent along with the created eve spend
+          and an xch spend that funds the transactions and pays fees.
+        - There is also an option to pass in a list of target puzzlehashes. If
+          provided this method will create an additional transaction transfering
+          the minted NFTs to the row-matched target.
+
+        :oaram metadata_list: A list of dicts containing the metadata for each NFT to be minted
+        :param target_list: [Optional] a list of targets for transfering minted NFTs (aka airdrop)
+        :param mint_number_start: [Optional] The starting point for mint number used in intermediate launcher
+        puzzle. Default: 1
+        :param mint_total: [Optional] The total number of NFTs being minted
+        :param xch_coins: [Optional] For use with bulk minting to provide the coin used for funding the minting spend.
+        This coin can be one that will be created in the future
+        :param xch_change_ph: [Optional] For use with bulk minting, so we can specify the puzzle hash that the change
+        from the funding transaction goes to.
+        :param new_innerpuzhash: [Optional] The new inner puzzle hash for the DID once it is spent. For bulk minting we
+        generally don't provide this as the default behaviour is to re-use the existing inner puzzle hash
+        :param did_coin: [Optional] The did coin to use for minting. Required for bulk minting when the DID coin will
+        be created in the future
+        :param did_lineage_parent: [Optional]  The  parent coin to use for the lineage proof in the DID spend. Needed
+        for bulk minting when the coin will be created in the future
```

Comment:
When will the DID coin be created in the future?  You can't mint any of these without the DID to create them.

Reply:
when we're creating a set of spend bundles offline for bulk minting, the DID gets spent once for each batch of minted NFTs. When we create the next spend bundle we have to pass in the right parent info, otherwise the built-in function to construct the lineage fails